python/test/test_core.py

```diff
@@ -192,10 +192,34 @@ def test_new_domain():
 def test_user_domain():
     from datetime import datetime
 
-    domain = dp.user_domain("all datetimes", lambda x: isinstance(x, datetime))
-    assert str(domain) == 'UserDomain("all datetimes")'
-    assert domain.member(datetime.now())
+    def datetime_domain(months=None):
+        """The domain of datetimes, restricted by user-defined months"""
+        if months is None:
+            months = set(range(1, 13))  # any month
+
```

Comment:
```suggestion
    def datetime_domain(months):
        """A domain of datetimes, restricted to the specified months"""
```
We don't need default behavior for the test, so simplify.